### ISSUE TITLE
Fisherman?

### ISSUE THREAD

**Issue by bucaran** [andrewrynhard/plugin-rustup#1]
@andrewrynhard 

Hey Andrew 

I was wondering if you were familiar with [fisherman](http://why.fisherman.sh)?

Some known issues with oh-my-fish:

* poor support / tons of unresolved issues
* slow shell startup / usage
* lack of powerful features 
* fewer users

**Comment by andrewrynhard** [comment]
I was just looking at it. Installed it but no `fisher` binary in sight. Where is `fisher` installed to?

**Comment by bucaran** [comment]
Do you mean the installer? 

https://github.com/fisherman/fisherman/blob/master/bin/install

**Comment by andrewrynhard** [comment]
I ran `curl -sL get.fisherman.sh | fish` but:
````
fisher
fish: Unknown command 'fisher'
````

**Comment by andrewrynhard** [comment]
I am running it on a headless server over ssh. Don't know if that matters?

**Comment by bucaran** [comment]
You have curl right?
<img width="612" alt="screen shot 2016-04-19 at 4 46 45 pm" src="https://cloud.githubusercontent.com/assets/8317250/14631380/095abafa-064e-11e6-9c4f-7b1fadfae02d.png">

**Comment by bucaran** [comment]
> I am running it on a headless server over ssh.

You can install it manually by cloning the repo and then running `./bin/install`, but if you have a particular use case, and it seems you do, I think fin may fit you better instead.

https://github.com/fishery/fin/

**Comment by bucaran** [comment]
@andrewrynhard 

> I am running it on a headless server over ssh.

You can install it manually by cloning the repo and then running `./bin/install`, but if you have a particular use case, and it seems you do, I think fin may fit you better instead.

https://github.com/fishery/fin/

**Comment by andrewrynhard** [comment]
Strange. I removed `~/.config/fisherman` and `~/.local/share/fisherman`, reinstalled and now it works.

**Comment by bucaran** [comment]
@andrewrynhard Okay. Mmm a corner case I did not anticipate in the installer, it seems.

**Comment by andrewrynhard** [comment]
How does fin differ from fisherman?

**Comment by bucaran** [comment]
fin is a rewrite, more lightweight, feature wise is the same at the end of the day (but there are some differences wrt to key bindings and how the bundle file works), and it's just one file, so to install it you just drag and drop the function into your fish dir and that's it.

**Comment by andrewrynhard** [comment]
Sweet. I will give it a try. Will fisherman be phased out and move towards fin? I don't want to configure for fisherman and migrate later? If fin will be the focus I would rather go with that. I am looking for a fish framework to write plugins for and I want to stick with one.

**Comment by bucaran** [comment]
Not at all, they both serve different use cases. I find fin works better for me because of how I do dotfiles, but I think most people are on fisherman these days, plus fisherman has a different config/cache separation, builtin search, index and more advanced completions.

**Comment by andrewrynhard** [comment]
Ok sounds good. I will play with them and see what I like. Really quickly, the dotfile manager I created, [alterant](https://github.com/autonomy/alterant.git), uses symlinks. Will fin or fisherman overwrite the link on an update to them?

**Comment by bucaran** [comment]
I think fin will work better for you then. Similar to OMF, fisherman does stuff to your config.fish. The first one makes a copy and creates a new one, the latter merges them.

fin does neither. You don't need to modify your config.fish to use fin, and it's just one function/file.

**Comment by andrewrynhard** [comment]
Great, thanks!

**Comment by bucaran** [comment]
One more word, this plugin uses an init.fish file. This is an omf only mechanism to run code on startup, that I must plead guilty originated myself with [wahoo](https://github.com/bucaran/wahoo). At the time it seemed great, but it's not a recommended practice anymore.

fish 2.3.0 will introduce [conf.d support](https://github.com/fish-shell/fish-shell/pull/2498) and both fin and fisherman are already on board this.

**Comment by bucaran** [comment]
One more word, this plugin uses an init.fish file. This is an omf-only hack to run code on startup, that I must plead guilty to have originated myself with [wahoo](https://github.com/bucaran/wahoo). At the time it seemed great, but it's not a recommended practice anymore.

fish 2.3.0 will introduce [conf.d support](https://github.com/fish-shell/fish-shell/pull/2498) and both fin and fisherman are already on board with this.

>  I am looking for a fish framework to write plugins for and I want to stick with one.

If you want to chat about this with me and the other members you are welcome to http://join.fisherman.sh

**Comment by andrewrynhard** [comment]
Thanks for the info. Just to clarify, I do not mind if fisherman does stuff to my config.fish, I would like to know if fisherman will write to the path that the symlink points to?

**Comment by andrewrynhard** [comment]
I just made a PR for omf that resolves `bundle` if it is a symlink. I want to know if I can do the same for fisherman if need be.

**Comment by bucaran** [comment]
> Thanks for the info. Just to clarify, I do not mind if fisherman does stuff to my config.fish, I would like to know if fisherman will write to the path that the symlink points to?

It will https://github.com/fisherman/fisherman/blob/master/bin/install#L152

**Comment by andrewrynhard** [comment]
Ok. I will give them a shot and see if I need to make a PR. Thanks for the information about these. They look great.

**Comment by bucaran** [comment]
Thanks, you are welcome to join the org as well https://github.com/fisherman

It would be a shame to waste your time and energy in a locked-in project like omf, which is not compat with fisherman stuff and does not want to open to change.

**Comment by andrewrynhard** [comment]
That sounds good. Send the invite and I will join.
